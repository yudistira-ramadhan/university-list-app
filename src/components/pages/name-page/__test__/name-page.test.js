import React from 'react'
import ReactDOM from 'react-dom'
import NamePage from '../name-page'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<NamePage/>, div)
})

it('renders NamePage correctly', () => {
  const { getByTestId } = render(<NamePage />)
  expect(getByTestId('name-page')).toBeTruthy()
})

it('matches snapshot', () => {
  const tree = renderer.create(<NamePage />).toJSON()
  expect(tree).toMatchSnapshot()
})