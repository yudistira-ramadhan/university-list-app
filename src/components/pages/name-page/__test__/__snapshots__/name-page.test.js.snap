// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="name-page"
>
  <div
    className="text-center jumbotron"
  >
    <h2
      className="font-weight-bolder"
    >
      University List by Name
    </h2>
    <hr
      className="my-4"
    />
    <p
      className="lead"
    >
      You can search the University from around the World.
    </p>
    <div
      className="text-center"
    >
      <div
        className="d-flex"
      >
        <input
          className="mr-2 form-control"
          name="university"
          onChange={[Function]}
          placeholder="Input university name..."
          type="text"
        />
        <button
          aria-label={null}
          className="btn btn-outline-primary"
          data-testid="search-button"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</div>
`;
